bokehjs/src/lib/models/tools/edit/edit_tool.ts

```diff
@@ -53,6 +53,17 @@ export abstract class EditToolView extends GestureToolView {
     cds.selection_manager.clear();
   }
 
+  _pop_glyphs(cds: ColumnarDataSource, num_objects: number, xkey: string | null, ykey: string | null) {
+    // Pops rows in the CDS until only num_objects are left
+    while (num_objects &&
+           ((xkey && cds.get_array(xkey).length >= num_objects) ||
+            (ykey && cds.get_array(ykey).length >= num_objects))) {
+      for (const column of cds.columns()) {
+        cds.get_array(column).splice(0, 1)
```

Comment:
Why the outer loop? wouldn't this work
```
N = cds.get_array(xkey).length = num_objects
cds.get_array(column).splice(0, N)
```
Also AFAIK Typed arrays do not have a `splice` merthod

Reply:
> Why the outer loop? Can't you just compute the number to delete up front?

Good point.

>Also AFAIK Typed arrays do not have a splice method

This is true, the other code already converts typed arrays in some cases, probably have to do that here too.